test/distributed/org/apache/cassandra/fuzz/harry/integration/dsl/HistoryBuilderOverridesTest.java

```diff
@@ -0,0 +1,376 @@
+            {
+                history.visitPartition(1,
+                                       (e) -> schema.ckGenerator.inflate(rng.next()))
+                       .insert(i);
+            }
+
+            for (int i = 0; i < partitionSize; i++)
+            {
+                history.visitPartition(1)
+                       .deleteRow(i);
+
+                visitor.replayAll();
+                history.validateAll(tracker, sut);
+            }
+        }
+    }
+
+
+    @Test
+    @SuppressWarnings("rawtypes")
+    public void regularAndStaticOverrideTest() throws Throwable
+    {
+        for (ColumnSpec.DataType type : new ColumnSpec.DataType[]{ ColumnSpec.asciiType(2, 0), ColumnSpec.int64Type })
+        {
+            SchemaSpec schema = new SchemaSpec("harry",
+                                               "test_overrides",
+                                               Arrays.asList(ColumnSpec.pk("pk1", ColumnSpec.asciiType(4, 10))),
+                                               Arrays.asList(ColumnSpec.ck("ck1", type, false)),
+                                               Arrays.asList(ColumnSpec.regularColumn("regular1", ColumnSpec.asciiType(2, 2)),
+                                                             ColumnSpec.regularColumn("regular2", ColumnSpec.int64Type)),
+                                               Arrays.asList(ColumnSpec.staticColumn("static1", ColumnSpec.asciiType(2, 2)),
+                                                             ColumnSpec.staticColumn("static2", ColumnSpec.int64Type)
+                                               ));
+
+            Map<String, Bijections.Bijection<?>> reGenerators = new HashMap<>();
+            reGenerators.put("regular1", ColumnSpec.asciiType(4, 4).generator());
+            reGenerators.put("regular2", ColumnSpec.int64Type.generator());
+            reGenerators.put("static1", ColumnSpec.asciiType(8, 4).generator());
+            reGenerators.put("static2", ColumnSpec.int64Type.generator());
+
+            DataTracker tracker = new DefaultDataTracker();
+            beforeEach();
+            sut.schemaChange(schema.compile().cql());
+
+            TokenPlacementModel.ReplicationFactor rf = new TokenPlacementModel.SimpleReplicationFactor(1);
+
+            int partitionSize = 100;
+            HistoryBuilder history = new HistoryBuilder(seed, partitionSize, 10, schema, rf);
+            ReplayingVisitor visitor = history.visitor(tracker, sut, SystemUnderTest.ConsistencyLevel.ALL);
+            EntropySource rng = new JdkRandomEntropySource(seed);
+
+            Map<String, Set<Object>> perColumnOverrides = new HashMap<>();
+            for (ColumnSpec column : schema.regularColumns)
+            {
+                perColumnOverrides.put(column.name, new HashSet<>());
+                for (int i = 0; i < partitionSize; i++)
+                {
+                    Object override = reGenerators.get(column.name).inflate(rng.next());
```

Comment:
Say I was testing SAI here and I wanted to try something like an empty string...

```
Object override = column.name.equals("regular1") ? "" : reGenerators.get(column.name).inflate(rng.next());
```

This doesn't seem to quite work when I try it...

Reply:
So you simply would do 

```
history.valueOverrides().override("regular1", 12345, "");
```

And then somewhere in your test you do

```
                    history.visitPartition(pdIdx)
                           .insert(i, new long[]{ 12345, 12345 });
```

Reply:
Added some docs!